### PR TITLE
[16.0][MIG] website_sale_product_attachment: Migration to version 16.0

### PR DESCRIPTION
**Pull request by ernesto-garcia-tecnativa**
@tecnativa TT44231

### CODE REVIEW

File: website_sale_product_attachment/tests/test_tour.py [hunk 1]
```diff
@@ -1,12 +1,13 @@
 # License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl)
 # Copyright 2021 Tecnativa - Víctor Martínez
 
-from odoo.tests import common
+import odoo.tests
```
Comment (chienandalu) [inline website_sale_product_attachment/tests/test_tour.py]:
```suggestion
from odoo.tests import tagged, HttpCase
```

File: website_sale_product_attachment/tests/test_tour.py [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
 
 
-class TestWebsiteSaleProductAttachmentTourl(common.HttpCase):
-    def setUp(self):
-        super().setUp()
+@odoo.tests.tagged("post_install", "-at_install")
+class TestWebsiteSaleProductAttachmentTourl(odoo.tests.HttpCase):
```
Comment (chienandalu) [inline website_sale_product_attachment/tests/test_tour.py]:
```suggestion
@tagged("post_install", "-at_install")
class TestWebsiteSaleProductAttachmentTourl(HttpCase):
```

File: website_sale_product_attachment/tests/test_tour.py [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
+    def setUpClass(self):
+        super().setUpClass()
```
Comment (chienandalu) [inline website_sale_product_attachment/tests/test_tour.py]: This change is quite useless sin we only have a single method

**Review by chienandalu — COMMENTED** [review]
In my functional test the attachments didn't show up for the public user (ie: not logged in)

**Comment by ernesto-garcia-tecnativa** [comment]
> In my functional test the attachments didn't show up for the public user (ie: not logged in)

I tried and works fine, you sure that field "Is public document" has marked into attachment file?

Example...
Captura de pantalla 2023-08-21 a la(s) 1 55 37 a m

**Comment by chienandalu** [comment]
> > In my functional test the attachments didn't show up for the public user (ie: not logged in)
> 
> I tried and works fine, you sure that field "Is public document" has marked into attachment file?

After your last changes it's the opposite. It shows up for public users but it doesn't for logged in users.

**Comment by ernesto-garcia-tecnativa** [comment]
@chienandalu because the website_attachment_ids field of product.template have a dynamic domain with the filter ```("public", "=", True)``` and others that filter internal files of Odoo

Please see this link:
https://github.com/OCA/e-commerce/pull/829/files#diff-e62ec0c8b3609baae7fb387ed2a30dc6a94ce7e7f5fe3ddb509127143856332cR27

**Comment by chienandalu** [comment]
> because the website_attachment_ids field of product.template have a dynamic domain with the filter `("public", "=", True)` and others that filter internal files of Odoo

But that's not my complain. I know that the attachment has to be public. The issue was that only the public user could see the dropdown in the product page and that's not the behavior in previous versions

**Review by chienandalu — APPROVED** [review]
Tested in local :+1:

**Comment by chienandalu** [comment]
please review @CarlosRoca13

**Review by CarlosRoca13 — APPROVED** [review]
Code and functional review :+1:

**Pull request merged**